src/Swarm/Game/Challenge.hs

```diff
@@ -57,7 +59,7 @@ data Challenge = Challenge
   { _challengeName :: Text
   , _challengeSeed :: Maybe Int
   , _challengeEntities :: EntityMap
-  , _challengeWorld :: WorldFun Int Entity
+  , _challengeWorld :: EntityMap -> WorldFun Int Entity
```

Comment:
I don't understand the purpose of introducing an `EntityMap` argument here.  That is what the whole `FromJSONE` infrastructure is supposed to be taking care of.  An `EntityMap` will be available at the time we read a challenge, so we can immediately use it to build the `WorldFun`.